cmd/protecodeExecuteScan.go

```diff
@@ -0,0 +1,317 @@
+
+	var resultFilePath string
+
+	if len(path) <= 0 {
+		resultFilePath = filepath.Join(cachePath, fileName)
+		if len(resultFilePath) <= 0 {
+			log.Entry().Fatalf("Protecode scan failed, there is no file path configured  : %v (filename:%v, PSPath: %v)", path, fileName, image.FSPath)
+		}
+	}
+
+	return fileName, resultFilePath
+}
+
+func tarImageData(tarFileName string, image pkgutil.Image, cacheImagePath string) {
+	tarFile, err := os.Create(tarFileName)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Protecode scan failed, error during create tar for the docker image")
+	}
+	if err := os.Chmod(tarFileName, 0644); err != nil {
+		log.Entry().WithError(err).Fatal("Protecode scan failed, error during create tar for the docker image")
+	}
+	defer tarFile.Close()
+
+	reference, err := name.ParseReference(image.Digest.String(), name.WeakValidation)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Protecode scan failed, not possible to parse reference of docker image")
+	}
+	err = tarball.Write(reference, image.Image, tarFile)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Protecode scan failed, error during create tar archive of docker image via tarball")
+	}
+
+	os.RemoveAll(cacheImagePath)
+}
+
+func getURLAndFileNameFromDockerImage(scanImage string, registryURL string) string {
+
+	completeURL := scanImage
+
+	if len(registryURL) > 0 {
+		if strings.HasSuffix(registryURL, "/") {
+			completeURL = fmt.Sprintf("remote://%v%v", registryURL, scanImage)
+		} else {
+			completeURL = fmt.Sprintf("remote://%v/%v", registryURL, scanImage)
+		}
+	}
+
+	if len(completeURL) <= 0 {
+		log.Entry().Fatal("Protecode scan failed, there is no scan image configured")
+	}
+
+	return completeURL
+}
+
+func executeProtecodeScan(client protecode.Protecode, config *protecodeExecuteScanOptions, fileName string, writeReportToFile func(resp io.ReadCloser, reportFileName string) error) (map[string]int, int) {
+
+	var parsedResult map[string]int = make(map[string]int)
+	//load existing product by filename
+	if config.Verbose {
+		log.Entry().Info("Protecode scan debug, load existing product")
+	}
+	productID := client.LoadExistingProduct(config.ProtecodeGroup, config.ReuseExisting)
+
+	// check if no existing is found or reuse existing is false
+	productID = uploadScanOrDeclareFetch(*config, productID, client, fileName)
+	if productID <= 0 {
+		log.Entry().Fatalf("Protecode scan failed, the product id is not valid (product id %v <= zero)", productID)
+	}
+	//pollForResult
+	if config.Verbose {
+		log.Entry().Info("Protecode scan debug, poll for scan result")
+	}
+	result := client.PollForResult(productID, config.ProtecodeTimeoutMinutes, config.Verbose)
+
+	jsonData, _ := json.Marshal(result)
+	ioutil.WriteFile("Vulns.json", jsonData, 0644)
+
+	//check if result is ok else notify
+	if len(result.Result.Status) > 0 && result.Result.Status == "F" {
+		log.Entry().Fatal("Protecode scan failed, please check the log and protecode backend for more details.")
```

Comment:
do we have the link to the backend here? If so, we could add it to the log message ...